Xamarin.Forms.Core/DataTemplate.cs

```diff
@@ -1,43 +1,51 @@
 using System;
 using System.Collections.Generic;
+using System.Threading;
+using Xamarin.Forms.Internals;
 
 namespace Xamarin.Forms
 {
-	public class DataTemplate : ElementTemplate
+	public class DataTemplate : ElementTemplate, IDataTemplateController
 	{
+		static int idCounter = 1;
+
+		int _id;
+		string _idString;
 		public DataTemplate()
 		{
+			_idString = GetType().FullName + idCounter++;
```

Comment:
Should this constructor also get the *Interlocked.Increment(ref idCounter);* treatment?

Reply:
fixed